test/es-module/test-esm-child-process-fork-main.mjs

```diff
@@ -0,0 +1,19 @@
+import '../common/index.mjs';
+import assert from 'assert';
+import { fork } from 'child_process';
+import { once } from 'events';
+
+if (process.argv[2] !== 'child') {
+  const currentFile = 'test-esm-child-process-fork-main.mjs';
+  const { pathname: filename } = new URL(currentFile, import.meta.url);
+  const cp = fork(filename, ['child']);
```

Comment:
Theoretically, all Node.js APIs are able to take a `URL` object instead of a path. It looks like `fork` has been forgotten, so I opened https://github.com/nodejs/node/pull/41225 to fix that. Would be OK waiting for the other PR to land first so we can simplify the test here?
```suggestion
  const cp = fork(new URL(import.meta.url), ['child']);
```

Reply:
What if we deliver this one and I improve it on the next one? My plan is to contribute more refining stuff such as coverage as well

Reply:
Fair enough. You'd need to convert the url to a path, and add `import { fileURLToPath } from 'url';` at the top level.
```suggestion
  const filename = fileURLToPath(new URL(import.meta.url));
  const cp = fork(filename, ['child']);
```